Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp

```diff
@@ -120,27 +111,41 @@ void UPSOverlayWidget::TickPlayFadeOverlayAnimation()
 // When a character has been changed current active progression row also changes
 void UPSOverlayWidget::OnCurrentRowDataChanged_Implementation(FPlayerTag PlayerTag)
 {
-	DisplayLevelUIOverlay();
-}
-
-void UPSOverlayWidget::SetOverlayItemsVisibility(ESlateVisibility VisibilitySlate)
-{
-	// Level is unlocked hide the blocking overlay
-	SetVisibility(VisibilitySlate);
+	const bool bIsNewCharacter = PreviousPlayerTagInternal != PlayerTag;
+	DisplayLevelUIOverlay(bIsNewCharacter);
+	PreviousPlayerTagInternal = PlayerTag;
```

Comment:
Please don't create a new property to store the previous player tag. Instead, determine whether it's a character switch or a skin change by following these steps:

1. Add a second `PreviousPlayerTag` argument to your `OnCurrentActiveSaveRowChanged` delegate.  
```
	DECLARE_DYNAMIC_MULTICAST_DELEGATE_TwoParams(FCurrentActiveSaveRowChanged, const FPlayerTag, NewPlayerTag, const FPlayerTag, PreviousPlayerTag);
```
2. In `UPSWorldSubsystem::SetCurrentRowByTag`, before assigning the new row, store the previous row in a local variable. This way, you can broadcast `OnCurrentActiveSaveRowChanged` with both the new and previous tags.

We should minimize property caching to reduce costs on their support and make future refactoring easier

Reply:
I agree, thanks https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/37ad4a3fced09d24a78d99c1a7b975b8b9caf3ae